awx/ui_next/src/screens/Host/HostAdd/HostAdd.jsx

```diff
@@ -23,6 +27,10 @@ function HostAdd() {
       const { data: response } = await HostsAPI.create(values);
       history.push(`${url}/${response.id}/details`);
     } catch (error) {
+      // check for field-specific errors from API
+      if (error.response?.data && typeof error.response.data === 'object') {
```

Comment:
This feels a little hacky to me... I don't know if there's a better way to see if the API error includes field-specific errors...?

Reply:
This looks fine and is probably about as good as it is going to get in terms of the type of checking we'll need to do to handle the field errors. There may be other structures we'll need to look out for, but this is the main one and the one we should start with.